### ISSUE TITLE
resettable property is not available in <SelectArrayInput>

### ISSUE THREAD

**Issue by IvanIvannov1952** [marmelab/react-admin#7718]
**What you were expecting:**
add resstable prop in SelectArrayInput

**Steps to reproduce:**
![image](https://user-images.githubusercontent.com/91272650/169473356-e7845a29-54a2-43c2-8b38-6153afe7859d.png)

**Related code:**

* Preferably, a CodeSandbox forked from 
 <SelectArrayInput
        {...props}
        source="groups"
        translateChoice
        resettable
        choices={segments}
    />

**Environment**

* React-admin version:4.1
* React version:17
* Browser: Brave

**Comment by slax57** [comment]
Thanks for submitting this

**Comment by davidhenley** [comment]
@djhi Why would you not support it? I'm so confused the decision on this.

**Comment by djhi** [comment]
Because of the following reasons:
- MUI does not support it either
- Having a clear button in addition of the dropdown icon is confusing
- We would have to save space between the selected items and the dropdown icon for the clear button

If you really need the clear button, I suggest you use the `AutocompleteArrayInput`

**Comment by MBach** [comment]
> 

Well, I disagree.

When you have a big list of predefined entries (around 15), with `SelectArrayInput`, you can quickly toggle up to 4-5 elements with the same amount of click. But it's kind of annoying to toggle again to restore the default listing. The `AutocompleteArrayInput` does not provide the same efficiency because you have to click and select for every item, so it's twice the same amount of clicks. And the menu closes every time, so when you do it very quickly, the screen is flashing.

**Comment by djhi** [comment]
Fair point. Can you open a PR so that we can iterate on the UI you suggest ?

**Comment by MBach** [comment]
> 

A Pull Request for suggesting a new prop?

**Comment by djhi** [comment]
No, a PR adding the reset button on the `SelectArrayInput` when `resetable` is set to `true`
